src/Cli.Tests/AddTelemetryTests.cs

```diff
@@ -0,0 +1,166 @@
+            Assert.IsNotNull(config);
+            Assert.IsNull(config.Runtime!.Telemetry);
+
+            // Add Telemetry
+            bool isSuccess = ConfigGenerator.TryAddTelemetry(
+                new AddTelemetryOptions(appInsightsConnString, isTelemetryEnabled, configPath),
+                new FileSystemRuntimeConfigLoader(fileSystem),
+                fileSystem);
+
+            // Assert after adding telemetry
+            Assert.AreEqual(expectSuccess, isSuccess);
+            if (expectSuccess)
+            {
+                Assert.IsTrue(fileSystem.FileExists(configPath));
+                Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(fileSystem.File.ReadAllText(configPath), out config));
+                Assert.IsNotNull(config);
+                Assert.IsNotNull(config.Runtime!.Telemetry);
+                TelemetryOptions telemetryOptions = config.Runtime!.Telemetry;
+                Assert.IsNotNull(telemetryOptions.ApplicationInsights);
+                Assert.AreEqual(isTelemetryEnabled is CliBool.True ? true : false, telemetryOptions.ApplicationInsights.Enabled);
+                Assert.AreEqual(appInsightsConnString, telemetryOptions.ApplicationInsights.ConnectionString);
+            }
+        }
+
+        /// <summary>
+        /// Test to verify when Telemetry section is present in the config
+        /// It should add application insights telemetry if telemetry section is empty
+        /// or overwrite the existing app insights telemetry with the given app insights telemetry options.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(true, DisplayName = "Add AppInsights Telemetry when telemetry section is empty.")]
+        [DataRow(false, DisplayName = "Overwrite AppInsights Telemetry when telemetry section already exists.")]
+        public void TestAddAppInsightsTelemetryWhenTelemetryAlreadyExists(bool isEmptyTelemetry)
+        {
+            MockFileSystem fileSystem = FileSystemUtils.ProvisionMockFileSystem();
+            string configPath = "test-app-insights-config.json";
+            string runtimeSection = isEmptyTelemetry ? RUNTIME_SECTION_WITH_EMPTY_TELEMETRY_SECTION : RUNTIME_SECTION_WITH_APP_INSIGHTS_TELEMETRY_SECTION;
+            string configData = $"{{{SAMPLE_SCHEMA_DATA_SOURCE},{runtimeSection}}}";
+            fileSystem.AddFile(configPath, new MockFileData(configData));
+
+            // Initial State
+            Assert.IsTrue(fileSystem.FileExists(configPath));
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(fileSystem.File.ReadAllText(configPath), out RuntimeConfig? config));
+            Assert.IsNotNull(config);
+            Assert.IsNotNull(config.Runtime);
+            Assert.IsNotNull(config.Runtime.Telemetry);
+
+            if (isEmptyTelemetry)
+            {
+                Assert.IsNull(config.Runtime!.Telemetry!.ApplicationInsights);
```

Comment:
nit: Since, `config.Runtime` and `config.Runtime.Telemetry` has been validated to be not null, check if it is possible to write this Assert without `!` operator.